featurebyte/routes/observation_table/controller.py

```diff
@@ -149,10 +153,52 @@ async def update_observation_table(
         data: ObservationTableUpdate
             ObservationTable update payload
 
+        Raises
+        ------
+        ObservationTableInvalidUseCaseError
+            if use_case_ids is provided in the payload or use_case_id_to_add/use_case_id_to_remove is invalid
+
         Returns
         -------
         Optional[ObservationTableModel]
         """
+
+        if data.use_case_ids:
+            raise ObservationTableInvalidUseCaseError("use_case_ids is not a valid field to update")
+
+        if data.use_case_id_to_add or data.use_case_id_to_remove:
+            observation_table = await self.observation_table_service.get_document(
+                document_id=observation_table_id
+            )
+            use_case_ids = observation_table.use_case_ids if observation_table.use_case_ids else []
+
+            # validate use case id to add
+            if data.use_case_id_to_add:
+                if data.use_case_id_to_add in use_case_ids:
+                    raise ObservationTableInvalidUseCaseError(
+                        f"Cannot add UseCase {data.use_case_id_to_add} as it is already associated with the ObservationTable."
+                    )
+                use_case = await self.use_case_service.get_document(
+                    document_id=data.use_case_id_to_add
+                )
+                if use_case.context_id != observation_table.context_id:
+                    raise ObservationTableInvalidUseCaseError(
+                        f"Cannot add UseCase {data.use_case_id_to_add} as its context_id is different from the existing context_id."
+                    )
```

Comment:
if `observation_table.context_id` is `None`, should we fail this part? Just to double check the behavior here.

Reply:
I think it make senses. An observation table cannot be attached to a UseCase unless it has been associated with a Context already